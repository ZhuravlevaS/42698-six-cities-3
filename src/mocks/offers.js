export default [
  {
    img: `img/apartment-01.jpg`,
    price: 120,
    rating: 4,
    description: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    isMarked: false,
    isPremium: true,
    id: 1
  },

  {
    img: `img/room.jpg`,
    price: 80,
    rating: 4,
    description: `Wood and stone place`,
    type: `Private room`,
    isMarked: true,
    isPremium: false,
    id: 2
  },

  {
    img: `img/apartment-02.jpg`,
    price: 132,
    rating: 4,
    description: `Canal View Prinsengracht`,
    type: `Apartment`,
    isMarked: false,
    isPremium: false,
    id: 3
  },

  {
    img: `img/apartment-03.jpg`,
    price: 180,
    rating: 5,
    description: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    isMarked: false,
    isPremium: false,
    id: 4
  },

  {
    img: `img/room.jpg`,
    price: 80,
    rating: 4,
    description: `Wood and stone place`,
    type: `Private room`,
    isMarked: false,
    isPremium: false,
    id: 5
  }
];
