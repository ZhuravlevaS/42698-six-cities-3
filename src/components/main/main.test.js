import React from 'react';
import Main from './main.jsx';
import renderer from 'react-test-renderer';
import offers from '../../mocks/dataCities.js';


it(`<Main/> render`, () => {
  const onMouseOver = jest.fn();
  const onMouseOut = jest.fn();
  const onCityClick = jest.fn();
  const saveCitiesData = jest.fn();
  const city = `Amsterdam`;

  const tree = renderer
    .create(<Main
      aparts={offers}
      onMouseOver={onMouseOver}
      onMouseOut={onMouseOut}
      city={city}
      onCityClick={onCityClick}
      saveCitiesData={saveCitiesData}/>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
