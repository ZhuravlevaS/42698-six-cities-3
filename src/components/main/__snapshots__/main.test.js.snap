// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> render 1`] = `
<div>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        id="icon-arrow-select"
        viewBox="0 0 7 4"
      >
        <path
          clipRule="evenodd"
          d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="icon-bookmark"
        viewBox="0 0 17 18"
      >
        <path
          d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 13 12"
      >
        <path
          clipRule="evenodd"
          d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>
  <div
    className="page page--gray page--main"
  >
    <header
      className="header"
    >
      <div
        className="container"
      >
        <div
          className="header__wrapper"
        >
          <div
            className="header__left"
          >
            <a
              className="header__logo-link header__logo-link--active"
            >
              <img
                alt="6 cities logo"
                className="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            className="header__nav"
          >
            <ul
              className="header__nav-list"
            >
              <li
                className="header__nav-item user"
              >
                <a
                  className="header__nav-link header__nav-link--profile"
                  href="#"
                >
                  <div
                    className="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    className="header__user-name user__name"
                  >
                    Oliver.conner@gmail.com
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      className="page__main page__main--index"
    >
      <h1
        className="visually-hidden"
      >
        Cities
      </h1>
      <div
        className="tabs"
      >
        <section
          className="locations container"
        >
          <ul
            className="locations__list tabs__list"
          >
            <li
              className="locations__item"
            >
              <a
                className="locations__item-link tabs__item "
                href="#"
                onClick={[Function]}
              >
                <span>
                  Paris
                </span>
              </a>
            </li>
            <li
              className="locations__item"
            >
              <a
                className="locations__item-link tabs__item "
                href="#"
                onClick={[Function]}
              >
                <span>
                  Cologne
                </span>
              </a>
            </li>
            <li
              className="locations__item"
            >
              <a
                className="locations__item-link tabs__item "
                href="#"
                onClick={[Function]}
              >
                <span>
                  Brussels
                </span>
              </a>
            </li>
            <li
              className="locations__item"
            >
              <a
                className="locations__item-link tabs__item tabs__item--active"
                href="#"
                onClick={[Function]}
              >
                <span>
                  Amsterdam
                </span>
              </a>
            </li>
            <li
              className="locations__item"
            >
              <a
                className="locations__item-link tabs__item "
                href="#"
                onClick={[Function]}
              >
                <span>
                  Hamburg
                </span>
              </a>
            </li>
            <li
              className="locations__item"
            >
              <a
                className="locations__item-link tabs__item "
                href="#"
                onClick={[Function]}
              >
                <span>
                  Dusseldorf
                </span>
              </a>
            </li>
          </ul>
        </section>
      </div>
      <div
        className="cities"
      >
        <div
          className="cities__places-container container"
        >
          <section
            className="cities__places places"
          >
            <h2
              className="visually-hidden"
            >
              Places
            </h2>
            <b
              className="places__found"
            >
              5
               places to stay in 
              Amsterdam
            </b>
            <withSortingVariants
              aparts={
                Array [
                  Object {
                    "bedrooms": 3,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "goods": Array [
                      "Heating",
                      "Kitchen",
                      "Cable TV",
                      "Washing machine",
                      "Coffee machine",
                      "Dishwasher",
                    ],
                    "host": Object {
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "id": 3,
                      "isPro": true,
                      "name": "Angelina",
                    },
                    "id": 1,
                    "images": Array [
                      "img/apartment-02.jpg",
                      "img/apartment-03.jpg",
                    ],
                    "isFavorite": false,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 52.370216,
                      "longitude": 4.8499948,
                      "zoom": 8,
                    },
                    "maxAdults": 4,
                    "previewImage": "img/apartment-03.jpg",
                    "price": 110,
                    "rating": 4,
                    "title": "Beautiful & luxurious studio at great location",
                    "type": "apartment",
                  },
                  Object {
                    "bedrooms": 3,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "goods": Array [
                      "Heating",
                      "Kitchen",
                      "Cable TV",
                      "Washing machine",
                      "Coffee machine",
                      "Dishwasher",
                    ],
                    "host": Object {
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "id": 4,
                      "isPro": true,
                      "name": "Angelina",
                    },
                    "id": 2,
                    "images": Array [
                      "img/apartment-02.jpg",
                      "img/apartment-03.jpg",
                    ],
                    "isFavorite": false,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 52.37514938,
                      "longitude": 4.6738799948,
                      "zoom": 8,
                    },
                    "maxAdults": 4,
                    "previewImage": "img/apartment-03.jpg",
                    "price": 10,
                    "rating": 4,
                    "title": "Beautiful & luxurious studio at great location",
                    "type": "apartment",
                  },
                  Object {
                    "bedrooms": 3,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "goods": Array [
                      "Heating",
                      "Kitchen",
                      "Cable TV",
                      "Washing machine",
                      "Coffee machine",
                      "Dishwasher",
                    ],
                    "host": Object {
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "id": 5,
                      "isPro": true,
                      "name": "Angelina",
                    },
                    "id": 3,
                    "images": Array [
                      "img/apartment-02.jpg",
                      "img/apartment-03.jpg",
                    ],
                    "isFavorite": false,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 52.35514938496378,
                      "longitude": 4.77537499948,
                      "zoom": 8,
                    },
                    "maxAdults": 4,
                    "previewImage": "img/apartment-03.jpg",
                    "price": 1200,
                    "rating": 3,
                    "title": "Beautiful & luxurious studio at great location",
                    "type": "apartment",
                  },
                  Object {
                    "bedrooms": 3,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.370216,
                        "longitude": 4.895168,
                        "zoom": 10,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "goods": Array [
                      "Heating",
                      "Kitchen",
                      "Cable TV",
                      "Washing machine",
                      "Coffee machine",
                      "Dishwasher",
                    ],
                    "host": Object {
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "id": 6,
                      "isPro": true,
                      "name": "Angelina",
                    },
                    "id": 4,
                    "images": Array [
                      "img/apartment-02.jpg",
                      "img/apartment-03.jpg",
                    ],
                    "isFavorite": false,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 52.35514938496378,
                      "longitude": 4.673877537499948,
                      "zoom": 8,
                    },
                    "maxAdults": 4,
                    "previewImage": "img/apartment-03.jpg",
                    "price": 120,
                    "rating": 1,
                    "title": "Beautiful & luxurious studio at great location",
                    "type": "apartment",
                  },
                  Object {
                    "bedrooms": 3,
                    "city": Object {
                      "location": Object {
                        "latitude": 48.85341,
                        "longitude": 2.3488,
                        "zoom": 10,
                      },
                      "name": "Paris",
                    },
                    "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                    "goods": Array [
                      "Heating",
                      "Kitchen",
                      "Cable TV",
                      "Washing machine",
                      "Coffee machine",
                      "Dishwasher",
                    ],
                    "host": Object {
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "id": 7,
                      "isPro": true,
                      "name": "Angelina",
                    },
                    "id": 8,
                    "images": Array [
                      "img/apartment-03.jpg",
                      "img/apartment-03.jpg",
                    ],
                    "isFavorite": false,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 48.85941,
                      "longitude": 2.34,
                      "zoom": 8,
                    },
                    "maxAdults": 4,
                    "previewImage": "img/apartment-02.jpg",
                    "price": 80,
                    "rating": 4.4,
                    "title": "Beautiful & luxurious studio at great location",
                    "type": "apartment",
                  },
                ]
              }
              city="Amsterdam"
              onMouseOut={[MockFunction]}
              onMouseOver={[MockFunction]}
            />
          </section>
          <div
            className="cities__right-section"
          >
            <section
              className="cities__map map"
              style={
                Object {
                  "backgroundImage": "none",
                }
              }
            >
              <div
                id="map"
                style={
                  Object {
                    "height": "100%",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
